File: ConferenceTask/ConferenceTask/Generator.cs
using System;
using System.IO;

namespace ConferenceTask
{
    class Generator
    {
        /// <summary>
        /// количество слушателей
        /// </summary>
        public const int Listeners = 1050;
        /// <summary>
        /// количество потоков
        /// </summary>
        public const int Sections = 3;
        /// <summary>
        /// количество докладов
        /// </summary>
        public const int Reports = 30;
        /// <summary>
        /// путь к файлу с входной матрицей
        /// </summary>
        public const string FilePath = "input.txt";

        private static readonly int[,] Matrix = new int[Reports,Listeners];

        public static readonly string[] Separator = new[]{" "};

        /// <summary>
        /// генерирует и записывает матрицу в файл
        /// </summary>
        public static void GenerateMatrix()
        {
            var random = new Random();
            using (var writer = new StreamWriter(FilePath))
            {
                for (int topic = 0; topic < Reports; topic++)
                {
                    for (int listener = 0; listener < Listeners; listener++)
                    {
                        writer.Write("{0}", random.Next(10) + Separator[0]);
                    }
                    writer.WriteLine();
                }
            }
        }

        /// <summary>
        /// считывает матрицу из файла
        /// </summary>
        /// <returns>
        /// входная матрица из файла
        /// </returns>
        public static int[,] ReadMatrixFromFile()
        {
            var matrix = new int[Reports, Listeners];
            using (var reader = new StreamReader(FilePath))
            {
                for  (int topic = 0; topic < Reports; topic++) 
                {
                    string line;
                    if ((line = reader.ReadLine()) == null) throw new Exception("Incorrect file!");
                    var inputArray = line.Split(Separator, StringSplitOptions.RemoveEmptyEntries);
                    if(inputArray.Length != Listeners) throw  new Exception("Incorrect file!");
                    for (int listener = 0; listener < Listeners; listener++)
                    {
                        int res;
                        int.TryParse(inputArray[listener], out res);
                        matrix[topic, listener] = res;
                    }
                }
            }
            return matrix;
        }

        public static string GenerateName(Random random)
        {
            var length = random.Next(3, 8);
            var name = "";
            for (var letter = 0; letter < length; letter++)
            {
                var ch = (char)('а' + random.Next(32));
                name += ch;
            }
            return name;
        }
    }
}

File: MainMAS/src/ConferenceTask/Generator.java
package ConferenceTask;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;
import java.util.StringTokenizer;

/**
 * Created with IntelliJ IDEA.
 * User: Andrew
 * Date: 18.01.14
 * Time: 16:58
 * To change this template use File | Settings | File Templates.
 */
public class Generator
{
    /**
     * количество слушателей
     */
    public static final int listeners = 10;
    /**
     * количество потоков
     */
    public static final int sections = 3;
    /**
     * количество докладов
     */
    public static final int reports = 30;
    /**
     * путь к файлу с входной матрицей
     */
    public static final String filePath = "input.txt";

    private static final int[][] matrix = new int[reports][listeners];

    public static final String separator = " ";

    /**
     * генерирует и записывает матрицу в файл
     */
    public static void generateMatrix ()
    {
        Random random = new Random();
        try
        {
            FileWriter writer = new FileWriter(filePath);
            for (int topic = 0; topic < reports; topic++)
            {
                for (int listener = 0; listener < listeners; listener++)
                {
                    writer.write(String.format("{0}", random.nextInt(10) + separator));
                }
                writer.write("\n");
            }
        }
        catch (IOException e)
        {
            e.printStackTrace();  //To change body of catch statement use File | Settings | File Templates.
        }
    }


    public static int[][] readMatrixFromFile (String fileName) throws IOException
    {
        if (fileName == null)
        {
            generateMatrix();
            fileName = filePath;
        }
        return readMatrixFromFile(new BufferedReader(new FileReader(fileName)));
    }

    /**
     * @return входная матрица из файла
     *         считывает матрицу из файла
     */
    public static int[][] readMatrixFromFile (BufferedReader input) throws IOException
    {
        String line = "";
        int matrix[][] = new int[reports][listeners];
        int lineNumber = 0;

        while ((line = input.readLine()) != null)
        {
            StringTokenizer tokenizer = new StringTokenizer(line, separator);

            int i = 0;

            while (tokenizer.hasMoreTokens())
            {
                String token = tokenizer.nextToken();
                if (token.isEmpty())
                {
                    continue;
                }
                matrix[lineNumber][i++] = Integer.parseInt(token);
            }
            lineNumber++;
        }

        return matrix;
    }


    //		public static String generateName(java.util.Random random)
    //		{
    //			int length = random.nextInt(8);
    //			String name = "";
    //			for (int letter = 0; letter < length; letter++)
    //			{
    //				char ch = (char)('а' + random.nextInt(32));
    //				name += ch;
    //			}
    //			return name;
    //		}
}